### PR TITLE
plan, executor: remove NeedHandle field

### PR DESCRIPTION
**Pull request by hanfei1991**
make code cleaner. @winoros

### CODE REVIEW

File: plan/logical_plan_builder.go [hunk 1]
```diff
@@ -1538,6 +1531,18 @@ func (b *planBuilder) buildTableDual() LogicalPlan {
 	return dual
 }
 
+func (ds *DataSource) getHandleSchemaCol() *expression.Column {
+	return &expression.Column{
+		FromID:   ds.id,
+		DBName:   ds.DBName,
+		TblName:  ds.tableInfo.Name,
+		ColName:  model.NewCIStr("_rowid"),
```
Comment (ngaut) [inline plan/logical_plan_builder.go]: Define as constant variable.

File: model/model.go [hunk 2]
```diff
@@ -155,6 +155,16 @@ func (t *TableInfo) GetPkColInfo() *ColumnInfo {
 	return nil
 }
 
+// GetHandleColInfo mocks a column info for extra handle column.
+func GetHandleColInfo() *ColumnInfo {
```
Comment (coocood) [inline model/model.go]: How about `NewExtraHandleColInfo`?

File: plan/logical_plan_builder.go [hunk 3]
```diff
@@ -1538,6 +1531,18 @@ func (b *planBuilder) buildTableDual() LogicalPlan {
 	return dual
 }
 
+func (ds *DataSource) getHandleSchemaCol() *expression.Column {
```
Comment (coocood) [inline plan/logical_plan_builder.go]: `newExtraHandleSchemaCol`

File: plan/physical_plan_builder.go [hunk 4]
```diff
@@ -731,16 +730,31 @@ func (p *DataSource) forceToIndexScan(idx *model.IndexInfo) PhysicalPlan {
 	return t.plan()
 }
 
+// If there is a table reader which needs to keep order, we should append a pk to table scan.
+func (ts *PhysicalTableScan) appendPK(ds *DataSource) {
```
Comment (coocood) [inline plan/physical_plan_builder.go]: The method name is not consistent with its behavior.
I think splitting it into two methods `hasHandleCol` and `appendExtraHandleCol` would be more clear.

File: executor/builder.go [hunk 5]
```diff
@@ -1103,12 +1077,13 @@ func buildNoRangeIndexLookUpReader(b *executorBuilder, v *plan.PhysicalIndexLook
 		desc:              is.Desc,
 		tableRequest:      tableReq,
 		columns:           is.Columns,
-		handleCol:         handleCol,
 		priority:          b.priority,
-		tableReaderSchema: tableReaderSchema,
 		dataReaderBuilder: &dataReaderBuilder{executorBuilder: b},
 		batchSize:         128,
 	}
+	if cols, ok := v.Schema().TblID2Handle[is.Table.ID]; ok {
```
Comment (winoros) [inline executor/builder.go]: If there is no handle column, `e.handleIndex` will be 0, will this do some bad effects?

Comment (hanfei1991) [reply executor/builder.go]: If there is no handle column, it indicates we needn't to sort.

Comment (winoros) [reply executor/builder.go]: We'd better add this into the comment  of `handleIndex`. It's used only when `e.keepOrder` is true.

Comment (hanfei1991) [reply executor/builder.go]: fixed.

**Pull request merged**